src/core/jit/model.rs

```diff
@@ -388,6 +420,22 @@ impl<Input> OperationPlan<Input> {
     pub fn size(&self) -> usize {
         self.flat.len()
     }
+
+    pub fn field_is_scalar<Extensions>(&self, field: &Field<Extensions, Input>) -> bool {
+        self.index.type_is_scalar(field.type_of.name())
```

Comment:
This information should be added to the field at the time of planning instead of dynamically performing a string comparison everytime.